File: Part 5 - Association Rule Learning/16 - Apriori/apriori.R
# Apriori

# Data preprocessing
# install.packages('arules')
library(arules)
dataset = read.transactions("Market_Basket_Optimisation.csv", sep = ',', rm.duplicates = TRUE)
summary(dataset, top)
itemFrequencyPlot(dataset, topN = 10)

# Training apriori on the dataset

# you have to set your minimum support and confidence based on your business problem and objective. it is subjective
rules = apriori(dataset, parameter = list(support = 0.0037, confidence =  0.2))

# Visualizing the results
inspect(sort(rules, by = 'lift')[1:10])

File: Part 5 - Association Rule Learning/16 - Apriori/apriori.py
#%% Apriori

import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Importing the dataset
dataset = pd.read_csv('Market_Basket_Optimisation.csv', header = None)
transactions = []
for i in range(len(dataset)):
    transactions.append([str(dataset.values[i, j]) for j in range(0, 20)])

#%% Training Apriori on the dataset
from apyori import apriori

# These are the items that are bought at least three times per day in a week.
# A very high minimal confidence probably is going to return only the most purchased items and probably there is not a real relation between them. Those items are just purchased a lot.
rules = apriori(transactions, min_support=3*7/len(transactions), min_confidence=0.2, min_lift=3, min_length = 2)

#%% Visualizing the results
# These rules are already sorted by the relevance. The relevance is a criteria confined by support, lift etc.
results = list(rules)